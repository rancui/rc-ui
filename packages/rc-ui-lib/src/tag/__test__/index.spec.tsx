import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { Tag } from '..';

describe('Tag', () => {
  afterEach(() => {
    cleanup();
  });

  it('should emit close event when clicking the close icon', async () => {
    const onClose = jest.fn();
    const { container } = render(<Tag closeable onClose={onClose} />);
    await fireEvent.click(container.querySelector('.rc-tag__close'));
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it('should hide tag when the show prop is false', () => {
    const { container } = render(<Tag visible={false} />);
    expect(container).toMatchSnapshot();
  });

  it('should not trigger click event when clicking the close icon', async () => {
    const onClick = jest.fn();
    const { container } = render(<Tag closeable onClick={onClick} />);
    await fireEvent.click(container.querySelector('.rc-tag__close'));
    expect(onClick).toHaveBeenCalledTimes(0);
    await fireEvent.click(container.querySelector('.rc-tag'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should render border-color correctly', () => {
    const { container } = render(<Tag plain color="red" textColor="blue" />);
    expect(container).toMatchSnapshot();
  });
});
