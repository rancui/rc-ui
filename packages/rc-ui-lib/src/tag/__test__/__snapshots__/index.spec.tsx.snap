// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should hide tag when the show prop is false 1`] = `<div />`;

exports[`Tag should render border-color correctly 1`] = `
<div>
  <span
    class="rc-tag rc-tag--plain rc-tag--default"
    style="color: blue; border-color: red;"
  />
</div>
`;
